src/helpers/format-chat-history.ts

```diff
@@ -83,7 +88,18 @@ async function createContextBlockSection(
   if (!issueNumber || isNaN(issueNumber)) {
     throw context.logger.error("Issue number is not valid");
   }
-  const prDiff = await fetchPullRequestDiff(context, org, repo, issueNumber);
+  const pulls = await fetchLinkedPrFromIssue(org, repo, issueNumber, context);
```

Comment:
These `pulls` rely on searching the repo for all PRs then parsing the PR bodies for a `#<issueNumber>`, which is recommended but not enforced so this isn't 100% reliable. We tend to use GQL API for this nowadays as it's most reliable. Also for queries that potentially return more than 100 items we should use `octokit.paginate(octokit.pulls.list))` as when scanning all PRs it's likely.

It's also relying on the PR author actually linking using `#` because if they copy paste the url which most do then it'll be a url in the pr body instead. I had logic for hashtag + url matching in my implementation but I think you may have removed it.

I think it may also be pulling more PR diffs than intended within this promise on each call to `createContextBlockSection`, unsure without QA-ing and inspecting the formatted chat on a context rich repo.